uid2_client/encryption.py

```diff
@@ -147,6 +149,86 @@ def _decrypt_token_v3(token_bytes, keys, now):
     return DecryptedToken(id_str, established, site_id, site_key.site_id)
 
 
+def _encrypt_token(uid2, identity_scope, master_key, site_key, site_id, now, token_expiry, ad_token_version):
+    site_payload = bytearray(128)
+    #Site id
+    site_payload[0:4] = int.to_bytes(site_id, byteorder='big', length=4)
+    site_payload[4:12] = int.to_bytes(0, byteorder='big', length=8)
+    site_payload[12:16] = int.to_bytes(0, byteorder='big', length=4)
+    site_payload[16:20] = int.to_bytes(0, byteorder='big', length=4)
+    site_payload[20:28] = int.to_bytes(int(now.timestamp())*1000, byteorder='big', length=8)
+    site_payload[36:] = bytes(base64.b64decode(uid2))
+
+    id_payload = _encrypt_gcm(bytes(site_payload), None, site_key.secret)
+
+    master_payload = bytearray(256)
+    master_payload[:8] = int.to_bytes(int(token_expiry.timestamp())*1000, byteorder='big', length=8)
+    master_payload[8:16] = int.to_bytes(int(now.timestamp()), byteorder='big', length=8)
+    master_payload[16:20] = int.to_bytes(0, byteorder='big', length=4)
+    master_payload[20:21] = int.to_bytes(1, byteorder='big', length=1)
+    master_payload[21:25] = int.to_bytes(0, byteorder='big', length=4)
+    master_payload[25:29] = int.to_bytes(0, byteorder='big', length=4)
+    master_payload[29:33] = int.to_bytes(site_key.key_id, byteorder='big', length=4)
+    master_payload[33:] = bytes(id_payload)
+
+    encrypted_master_payload = _encrypt_gcm(bytes(master_payload), None, master_key.secret)
+
+    root_writer = bytearray(len(encrypted_master_payload)+6)
+    first_char = uid2[0]
+    identity_type = IdentityType.Phone if first_char == 'F' or first_char == 'B' else IdentityType.Email
+    root_writer[0:1] = int.to_bytes((int(identity_scope) << 4 | int(identity_type) << 2), byteorder='big', length=1)
+    root_writer[1:2] = int.to_bytes(ad_token_version, byteorder='big', length=1)
+    root_writer[2:6] = int.to_bytes(master_key.key_id, byteorder='big', length=4)
+    root_writer[6:] = bytes(encrypted_master_payload)
+
+    if ad_token_version == AdvertisingTokenVersion.ADVERTISING_TOKEN_V4:
+        return Uid2Base64UrlCoder.encode(root_writer)
+
+    return base64.b64encode(root_writer)
+
+
+
+def encrypt_key(uid2, indentity_scope, keys, keyset_id=None, **kwargs):
```

Comment:
method names might be a bit misleading, encrypt_token and encrypt_key make it sound like we're encrypting a token or encrypting a key, rather than a raw UID